src/platforms/esp32/components/avm_sys/platform_nifs.c

```diff
@@ -513,6 +528,47 @@ static term nif_esp_deep_sleep_enable_gpio_wakeup(Context *ctx, int argc, term a
 }
 #endif
 
+static term nif_esp_gpio_wakeup_enable(Context *ctx, int argc, term argv[])
+{
+    UNUSED(ctx);
+    UNUSED(argc);
+
+    VALIDATE_VALUE(argv[0], term_is_integer);
+    if ((argv[1] != LOW_ATOM) && (argv[1] != HIGH_ATOM)) {
+        RAISE_ERROR(BADARG_ATOM);
+    }
+
+    avm_int_t gpio = term_to_int(argv[0]);
+    gpio_int_type_t int_type = (argv[1] == LOW_ATOM) ? GPIO_INTR_LOW_LEVEL : GPIO_INTR_HIGH_LEVEL;
+
+    esp_err_t ret = gpio_wakeup_enable(gpio, int_type);
+
+    return (ret == ESP_OK) ? OK_ATOM : ERROR_ATOM;
+}
+
+static term nif_esp_sleep_enable_gpio_wakeup(Context *ctx, int argc, term argv[])
+{
+    UNUSED(ctx);
+    UNUSED(argc);
+    UNUSED(argv);
+
+    esp_err_t ret = esp_sleep_enable_gpio_wakeup();
+
+    return (ret == ESP_OK) ? OK_ATOM : ERROR_ATOM;
+}
+
+static term nif_esp_sleep_enable_timer_wakeup_us(Context *ctx, int argc, term argv[])
```

Comment:
I would call this `esp:sleep_enable_timer_wakeup/1` to match esp-idf function name.

Reply:
That is a good discussion point.
- option1: `esp` module might map 1:1 esp-idf functions, with no other changes as much as possible.
- option2: we might do some opinionated improvements.

In this specific case I think it might be useful having a prefix with the time unit in the function name, since we have some kind of mix of function taking milliseconds (e.g. `deep_sleep(SleepMS)` and now microseconds).
On the other hand having 1:1 names makes super straightforward following esp-idf documentation while writing in erlang.

I'd like to hear your opinion.